src/cargo/core/workspace.rs

```diff
@@ -201,6 +201,17 @@ impl<'cfg> Workspace<'cfg> {
         }.parent().unwrap()
     }
 
+    /// Returns the root path of this workspace.
+    ///
+    /// That is, this returns the path of the directory containing the
+    /// `Cargo.toml` which is the root of this workspace.
+    pub fn workspace_dir(&self) -> Option<&Path> {
+        match self.root_manifest {
```

Comment:
Hm, there doesn't seem to be enough interest for this feature. I managed to bypass it for my needs. 

I am willing to work on it this weekend.